libGitWrap/Tree.cpp

```diff
@@ -81,19 +81,38 @@ namespace Git
         return PrivatePtr(new Tree::Private(d->repo(), subObject));
     }
 
-    DiffList Tree::diffToTree(Result& result , Tree newTree)
+    /**
+     * @brief           Diff this tree to a newer tree.
+     *
+     * @param[in,out]   result      A Result object; see @ref GitWrapErrorHandling
+     *
+     * @param[in]       newTree     is considered as the "new tree" to compare with
+     *
+     * @return          the differences as @ref DiffList object
+     *
+     * The calling Tree object is considered as the "old tree" and @p newTree
+     * is the "new tree" to compare with. When exchanging both trees, the
+     * result will be inversed.
+     *
+     * An invalid Tree object is is considered an empty tree.
```

Comment:
This depends on which of both tree's is empty (i.e. NULL). The usual case is the exact opposite from what you wrote :smile:. If the old tree (*this* object) is invalid, the resulting `DiffList` will contain all content of `newTree` as **additions**. This is important and actually the reason, why I wrote the comments.

Comparison works in both directions:
```c++
Git::Tree oldTree;   // this is our "empty" tree from the parent commit 
Git::Tree newTree = treeFromCommit; // this is the root-commit's tree

// this is how we call it
DiffList dl              = oldTree.diffToTree( newTree );
// everything from "newTree" will appear as added

DiffList dlReverse = newTree.diffToTree( oldTree ); 
// everything from "newTree" will appear as removed
```

Think I should add this example, too.